tests/passes/tdcc-duplicate/seq.expect

```diff
@@ -0,0 +1,53 @@
+import "primitives/core.futil";
+import "primitives/memories/comb.futil";
+component main(@go go: 1, @clk clk: 1, @reset reset: 1) -> (@done done: 1) {
+  cells {
+    a = std_reg(2);
+    b = std_reg(2);
+    c = std_reg(2);
+    @generated fsm0 = std_reg(2);
+    @generated fsm1 = std_reg(2);
+  }
+  wires {
+    group A {
+      a.in = 2'd0;
+      a.write_en = 1'd1;
+      A[done] = a.done;
+    }
+    group B {
+      b.in = 2'd1;
+      b.write_en = 1'd1;
+      B[done] = b.done;
+    }
+    group C {
+      c.in = 2'd2;
+      c.write_en = 1'd1;
+      C[done] = c.done;
+    }
+    group tdcc {
+      A[go] = !A[done] & fsm0.out == 2'd0 ? 1'd1;
+      B[go] = !B[done] & fsm0.out == 2'd1 ? 1'd1;
+      C[go] = !C[done] & fsm1.out == 2'd2 ? 1'd1;
+      fsm0.in = fsm0.out == 2'd0 & A[done] ? 2'd1;
+      fsm0.write_en = fsm0.out == 2'd0 & A[done] ? 1'd1;
+      fsm1.in = fsm0.out == 2'd0 & A[done] ? 2'd1;
+      fsm1.write_en = fsm0.out == 2'd0 & A[done] ? 1'd1;
+      fsm0.in = fsm0.out == 2'd1 & B[done] ? 2'd2;
+      fsm0.write_en = fsm0.out == 2'd1 & B[done] ? 1'd1;
+      fsm1.in = fsm0.out == 2'd1 & B[done] ? 2'd2;
+      fsm1.write_en = fsm0.out == 2'd1 & B[done] ? 1'd1;
+      fsm0.in = fsm0.out == 2'd2 & C[done] ? 2'd3;
+      fsm0.write_en = fsm0.out == 2'd2 & C[done] ? 1'd1;
+      fsm1.in = fsm0.out == 2'd2 & C[done] ? 2'd3;
+      fsm1.write_en = fsm0.out == 2'd2 & C[done] ? 1'd1;
+      tdcc[done] = fsm0.out == 2'd3 ? 1'd1;
```

Comment:
All of these internal transitions use `fsm0`. Is that intentional? 

I think it might be worth doing the following: all `fsm==0../n2` guards should read from `fsm0` and all `fsm==n/2....n` guards should read from `fsm1`. 

@parthsarkar17 lmk if you want to take a shot at this. I think I could also probably do this as well.

Reply:
Nvm, I'm impatient so I just did it myself. 

@parthsarkar17 please take a sec to review though to see if the changes look good to you.

Reply:
Yes, that is intentional. The enables (e.g. `B[go] = !B[done] & fsm{x}.out == 2'd2 ? 1'd1;`) are the ones that are split evenly between the two registers. The transitions for each of the registers are based on the logic for `fsm0`. I thought that was fine because the second register can just update based off of the first register's logic. If we were to split this logic, then that might just add more complexity. Is that right?

Reply:
So your idea is there are two choices: 

1) register A and register B both read exclusively from register A 
2) register A and register B both read from both registers A and B. 

And your idea is that (1) will use more resources/have a worse slack than (2). 

That may be the case. Here's the reason I think it might actually be more profitable to do (2). 
The number of queries (i.e., `fsm == i`) you have to do is actually fewer: 
In case (1) `fsm0` will probably have to perform N queries since it's in charge of all the transitions. `fsm1` will probably have to perform N/2 queries (for the 1/2 of the enable queries). Which leads to 1.5 N queries total. 
In case (2) `fsm0` and `fsm1` will both have to perform `N/2` queries (since all queries are split), which leads to N queries total. 

I'm not sure which is correct: it might be worth running some experiments to see. We can test this commit vs. a previous commit.

Reply:
Yeah those are also the options as I see them. I think resources wise, they should be the same (because the logic  for each state transition needs to come from one of the registers, doesn't matter from which, right?), but worst slack might be worse if we read exclusively from A. 

And yeah, I totally didn't think about the unevenness that could come from all transition reads coming from A. The math sounds good to me!